stage #4/module #5. AWS/theory/dynamodb.md

```diff
@@ -0,0 +1,143 @@
+
+DynamoDB also provides with a service that cashes information anf acts as an in-memory database. Amazon DynamoDB Accelerator (DAX) is a fully managed, highly available, in-memory cache for DynamoDB that delivers up to a 10x performance improvement.
+[Read more](https://aws.amazon.com/dynamodb/dax/)
+
+## Quiz
+
+1. Is DynamoDB a serverless solution?
+- a) Yes
+- b) No
+- c) Yes, but DAX which is a part of the service requires instances
+- d) Yes, but instances are used when replication is enabled
+2. There are several storage types available in DynamoDB. They are ...
+- a) Yes, several ones. HDD, SSD
+- b) Yes, several ones. HDD, cold HDD, SSD
+- c) No, HDD only
+- d) No, SSD only
+3. Each item should follow a particular schema while being stored in DynamoDB?
+- a) Yes
+- b) Only if SSD is used for storing
+- c) Only if DAX enabled needed
+- d) No
+4. Suppose you have an item with a concrete partition key. How will DynamoDB resolve the situation putting this item in the same partition?
+- a) An existing item will be replaced with a new one
+- b) An existing item will be overwritten with a new one
+- c) There will be two item within the same partition but with different sort keys
+- d) There will be two item within the same hash key and sort key
+5. Suppose you have to access an item, and you know its partition key and other attributes' values. What will be faster to run?
+- a) Scan since you know all the attributes' values
+- b) Query since you know its partition key
+- c) Query since you know all the attributes' values
+- d) Scan since you know its partition key
+6. Is it possible to modify an LSI after its creation?
+- a) No
+- b) Yes
+- c) Yes, if it has only one item added
+- d) Yes, if there is only on LSI within the table
+7. AWS DynamoDB calls can be throttled if ...
+- a) No items found with specified parameters
+- b) RCU or WCU not configured
+- c) RCU or WCU exceeded
+- d) They never be throttled
+8. Use DAX in case you want to improve ...
+- a) Read performance
+- b) Write performance
+- c) Read and write performance
+- d) ElastiCache is a better approach
+9. Consider you want to ger rid of outdated items in order to reduce cost. You'd better ...
+- a) Use TTL feature
+- b) Manually delete them
+- c) Read a table, detect outdated items, create a new table without outdated items and delete the old table
+- d) Use scan features to filter old items
+10. What is the best approach to handle occasionally overloaded errors?
+- a) ElastiCache to read cached data
+- b) DAX to read cached data
+- c) Exponential backoff
+- d) If an error happen then no need to try it again because DynamoDB provides with an ultimate resources, so no errors possible if data exists
+
+## General requirements
+1. A mentee should be able to explain all general purpose of the service
+2. A mentee should be able to answer all the questions during a demo session.
+
+## Keys
```

Comment:
@a1exanddrovich Let's not post keys here, it's an open repository.

Reply:
fixed